### ISSUE TITLE
Error: Failure code 0002

### ISSUE THREAD

**Issue by nicoduj** [mrbungle64/ecovacs-deebot.js#27]
Hi,

I am evelopping an homebridge plugin for deebots and there is an issue when you try to connect to the API twice from the same device, with consecutive attemps : I often get this error code on connection.

I have fixed this in the original sucks.js and did a pull request, but the maintainer is not there anymore and I would like to switch to your package but can't do any pull request on it since it is a fork also.

Here is the original pull request : https://github.com/jeremy-green/sucks.js/pull/1 

I use uniqid dependency, and changed the following line : 

```javascript
params['requestId'] = EcoVacsAPI.md5(Number.parseFloat(Date.now() / 1000).toFixed(0));
```
to
```javascript
params['requestId'] = EcoVacsAPI.md5(uniqid());
```

And then everything is good.

**Comment by mrbungle64** [comment]
> Hi,

Hi Nicolas,
> 
> I am evelopping an homebridge plugin for deebots and there is an issue when you try to connect to the API twice from the same device, with consecutive attemps : I often get this error code on connection.
> 
> I have fixed this in the original sucks.js and did a pull request, but the maintainer is not there anymore and I would like to switch to your package but can't do any pull request on it since it is a fork also.

I appreciate that you want to use this library for your project :+1: 
You are also welcome to participate in this project and create pull requests.

> 
> Here is the original pull request : [jeremy-green#1](https://github.com/jeremy-green/sucks.js/pull/1)
> 
> I use uniqid dependency, and changed the following line :
> 
> ```js
> params['requestId'] = EcoVacsAPI.md5(Number.parseFloat(Date.now() / 1000).toFixed(0));
> ```
> 
> to
> 
> ```js
> params['requestId'] = EcoVacsAPI.md5(uniqid());
> ```
> 
> And then everything is good.

Thank you! I already made some tests and it seems to finally fix this issue.
That's good news :+1: 

Do you want to create a pull request for this fix?

**Comment by nicoduj** [comment]
Thanks for taking a look into it ! I have to delete my fork of original repo to fork your,s will do I hope today. I would alos wonder why you did not use / populate a ready state ? I was using it in order to reconnect in case of disconnect (xmpp / mqtt).

**Comment by mrbungle64** [comment]
> Thanks for taking a look into it ! I have to delete my fork of original repo to fork yours will do I hope today.
> 
> I would also wonder why you did not use / populate a ready state like it was done in sucks.js ? I was using it in order to reconnect in case of disconnect (xmpp / mqtt).

ok, you did the implementation in [this commit](https://github.com/jeremy-green/sucks.js/pull/1/commits/a50c8052d02a3a7cafd0e1c49e26f5ef677c1c13), right? I'll have a look at it.

> 
> Last but not least, on my first migration attemps I don't get events on my object , looking into it before a complete switch :)

ok ;)

btw: which Deebot model are you using?

**Comment by nicoduj** [comment]
I am working on the pull request, will submit my whole changes once I have something working. I use a deebot ozmo slim 10. IT is a code matter, since if i set the event in my api class, i get it , but when i retrieve vacbots objects through an array and set it on "top level", I d'ont. I think it is some changes from the new Vacbot() from original impl and your getVacBot function.

**Comment by mrbungle64** [comment]
@nicoduj 

Is the Deebot Ozmo Slim 10 model using MQTT or XMPP as communication protocol?

**Comment by nicoduj** [comment]
MQTT

**Comment by nicoduj** [comment]
Made the pull request, figured out that in your lib the connection is made directly, was not the case, so my events where attached after and di not get the first one.

https://github.com/mrbungle64/ecovacs-deebot.js/pull/28

Thanks for the great work !

**Comment by boriswerner** [comment]
Great :-)

**Comment by nicoduj** [comment]
Can you tell me when you will publish on NPM ? i have bumped the verison in package.json. I will then be able to make my plugin depends on your package instead of my fork, it will prevent some installations errors ;p

**Comment by mrbungle64** [comment]
> Can you tell me when you will publish on NPM ? i have bumped the verison in package.json. I will then be able to make my plugin depends on your package instead of my fork, it will prevent some installations errors ;p

I'm planning to publish it within the next hour. ok?

**Comment by nicoduj** [comment]
Yes, definitely fine ! I have work to do still on my side, and don't expect to publish before tonight, so take your time :)

**Comment by mrbungle64** [comment]
@nicoduj  @boriswerner 
I just published version 0.4.5 on npm :)

**Comment by nicoduj** [comment]
thanks a lot !
